subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/serialization/codecs/LocalFileDependencyBackedArtifactSetCodec.kt

```diff
@@ -156,11 +167,39 @@ class LocalFileDependencyBackedArtifactSetCodec(
             val transforms = readNonNull<Map<ImmutableAttributes, MappingSpec>>()
             FixedArtifactVariantSelector(matchingOnArtifactFormat, transforms, NoOpTransformedVariantFactory)
         }
-        return LocalFileDependencyBackedArtifactSet(FixedFileMetadata(componentId, files), filter, selector, artifactTypeRegistry, calculatedValueContainerFactory)
+        return DeserializedLocalFileDependencyArtifactSet(
+            FixedFileMetadata(componentId, files),
+            filter,
+            selector,
+            artifactTypeRegistry,
+            calculatedValueContainerFactory,
+            allowNoMatchingVariants
+        )
     }
 }
 
 
+private
+class DeserializedLocalFileDependencyArtifactSet(
```

Comment:
💅  It's `private`, yes, but it's also still a new top-level type in a non-internal package and should get a docstring comment.

Reply:
The question is... why is this in a non-internal package?

@gradle/bt-configuration Shouldn't all these codecs be internal? These classes show up as IDE hints when writing buildscripts. That seems wrong to me

Reply:
Added some doc